Sorry for the delay, there's now an `@coral-xyz/borsh` package with version `0.2.6` that includes the u256 changes. I'm trying to make another PR that switches over from using `project-serum/borsh` to `coral-xyz/borsh` but running into random typescript errors that I'm not very familiar with. Namely `Cannot find module '@coral-xyz/borsh' or its corresponding type declarations.` Which might be as a result of this `warning " > @coral-xyz/borsh@0.2.6" has unmet peer dependency "@solana/web3.js@^1.2.0".`. Anyway, if you want to have a stab at updating the package used across anchor to the coral-xyz package that would be appreciated but no worries not.

--- a/lang/syn/src/idl/mod.rs
+++ b/lang/syn/src/idl/mod.rs
@@ -195,6 +195,8 @@ pub enum IdlType {
     F64,
     U128,
     I128,
+    U256,
+    I256,
     Bytes,
     String,
     PublicKey,

--- a/ts/packages/anchor/src/coder/borsh/idl.ts
+++ b/ts/packages/anchor/src/coder/borsh/idl.ts
@@ -51,6 +51,12 @@ export class IdlCoder {
       case "i128": {
         return borsh.i128(fieldName);
       }
+      case "u256": {
+        return borsh.u256(fieldName);
+      }
+      case "i256": {
+        return borsh.i256(fieldName);
+      }
       case "bytes": {
         return borsh.vecU8(fieldName);
       }

--- a/ts/packages/anchor/src/coder/common.ts
+++ b/ts/packages/anchor/src/coder/common.ts
@@ -58,6 +58,10 @@ function typeSize(idl: Idl, ty: IdlType): number {
       return 16;
     case "i128":
       return 16;
+    case "u256":
+      return 32;
+    case "i256":
+      return 32;
     case "bytes":
       return 1;
     case "string":

--- a/ts/packages/anchor/src/idl.ts
+++ b/ts/packages/anchor/src/idl.ts
@@ -121,6 +121,8 @@ export type IdlType =
   | "f64"
   | "u128"
   | "i128"
+  | "u256"
+  | "i256"
   | "bytes"
   | "string"
   | "publicKey"

--- a/ts/packages/anchor/src/program/namespace/types.ts
+++ b/ts/packages/anchor/src/program/namespace/types.ts
@@ -121,7 +121,7 @@ type TypeMap = {
 } & {
   [K in "u8" | "i8" | "u16" | "i16" | "u32" | "i32" | "f32" | "f64"]: number;
 } & {
-  [K in "u64" | "i64" | "u128" | "i128"]: BN;
+  [K in "u64" | "i64" | "u128" | "i128" | "u256" | "i256"]: BN;
 };
 
 export type DecodeType<T extends IdlType, Defined> = T extends keyof TypeMap

--- a/ts/packages/anchor/tests/coder-types.spec.ts
+++ b/ts/packages/anchor/tests/coder-types.spec.ts
@@ -44,7 +44,7 @@ describe("coder.types", () => {
     assert.deepEqual(coder.types.decode("MintInfo", encoded), mintInfo);
   });
 
-  test('Can encode and decode 256-bit integers', () => {
+  test("Can encode and decode 256-bit integers", () => {
     const idl = {
       version: "0.0.0",
       name: "basic_0",
@@ -82,6 +82,9 @@ describe("coder.types", () => {
 
     const coder = new BorshCoder(idl);
     const encoded = coder.types.encode("IntegerTest", testing);
-    assert.strictEqual(coder.types.decode("IntegerTest", encoded).toString(), testing.toString());
+    assert.strictEqual(
+      coder.types.decode("IntegerTest", encoded).toString(),
+      testing.toString()
+    );
   });
 });

--- a/ts/packages/anchor/tests/coder-types.spec.ts
+++ b/ts/packages/anchor/tests/coder-types.spec.ts
@@ -1,5 +1,6 @@
 import * as assert from "assert";
 import { BorshCoder } from "../src";
+import BN from "bn.js";
 
 describe("coder.types", () => {
   test("Can encode and decode user-defined types", () => {
@@ -41,5 +42,46 @@ describe("coder.types", () => {
     const encoded = coder.types.encode("MintInfo", mintInfo);
 
     assert.deepEqual(coder.types.decode("MintInfo", encoded), mintInfo);
+  });
+
+  test('Can encode and decode 256-bit integers', () => {
+    const idl = {
+      version: "0.0.0",
+      name: "basic_0",
+      instructions: [
+        {
+          name: "initialize",
+          accounts: [],
+          args: [],
+        },
+      ],
+      types: [
+        {
+          name: "IntegerTest",
+          type: {
+            kind: "struct" as const,
+            fields: [
+              {
+                name: "unsigned",
+                type: "u256" as const,
+              },
+              {
+                name: "signed",
+                type: "i256" as const,
+              },
+            ],
+          },
+        },
+      ],
+    };
+
+    const testing = {
+      unsigned: new BN(2588012355),
+      signed: new BN(-93842345),
+    };
+
+    const coder = new BorshCoder(idl);
+    const encoded = coder.types.encode("IntegerTest", testing);
+    assert.strictEqual(coder.types.decode("IntegerTest", encoded).toString(), testing.toString());
   });
 });

